server/publications/collections/product-publications.app-test.js

```diff
@@ -99,50 +130,58 @@ describe("Publication", function () {
     });
 
     describe("Products", function () {
-      it("should return all products to admins", function (done) {
+      it("should return all products to admins in the Primary Shop", function (done) {
         // setup
-        sandbox.stub(Reaction, "getShopId", () => shopId);
+        sandbox.stub(Reaction, "getShopId", () => primaryShopId);
         sandbox.stub(Roles, "userIsInRole", () => true);
         sandbox.stub(Reaction, "hasPermission", () => true);
-        sandbox.stub(Reaction, "getShopsWithRoles", () => [shopId, primaryShopId]);
-
-        let isDone = false;
+        sandbox.stub(Reaction, "getShopsWithRoles", () => [shopId, merchantShopId, primaryShopId]);
 
         collector.collect("Products", 24, undefined, {}, (collections) => {
-          const products = collections.Products;
-
-          expect(products.length).to.equal(4);
+          const productIds = collections.Products.map(p => p._id);
 
-          if (!isDone) {
-            isDone = true;
-            done();
-          }
-        });
+          expect(productIds).to.eql(allProductIds);
+        }).then((_collections) => done(), done);
       });
 
-      it("returns products from only the shops for which an admin has createProduct Role", function (done) {
+      it("should return all products from the current shop to admins in a Merchant Shop", function (done) {
         // setup
         sandbox.stub(Reaction, "getShopId", () => shopId);
         sandbox.stub(Roles, "userIsInRole", () => true);
         sandbox.stub(Reaction, "hasPermission", () => true);
-        sandbox.stub(Reaction, "getShopsWithRoles", () => [primaryShopId]);
+        sandbox.stub(Reaction, "getShopsWithRoles", () => [shopId, merchantShopId, primaryShopId]);
 
-        let isDone = false;
+        // let isDone = false;
 
         collector.collect("Products", 24, undefined, {}, (collections) => {
-          const products = collections.Products;
+          const productIds = collections.Products.map(p => p._id);
 
-          expect(products.length).to.equal(1);
+          expect(productIds).to.eql(merchantShop1ProductIds);
+        }).then((_collections) => done(), done);
+      });
 
-          if (!isDone) {
-            isDone = true;
-            done();
-          }
-        });
+      it("returns products from only the shops for which an admin has createProduct Role", function (done) {
+        // setup
+        sandbox.stub(Reaction, "getShopId", () => primaryShopId);
+        sandbox.stub(Roles, "userIsInRole", () => true);
+        sandbox.stub(Reaction, "hasPermission", () => true);
+        sandbox.stub(Reaction, "getShopsWithRoles", () => [shopId]);
+
+        collector.collect("Products", 24, undefined, {}, (collections) => {
+          const productIds = collections.Products.map(p => p._id);
+
+          expect(productIds).to.eql(merchantShop1ProductIds);
+        }).then((_collections) => done(), done);
```

Comment:
not terribly important note:
the lambda is required to be sure that `done` is called with no parameters, indicating success
in addition, the underscore in `_collections` is a Ruby convention to say "this param is ignored". I included it because I didn't want a future refactorer to think, "`() => done()` is the same as `done`, let's change it". making it `.then(done, done)` would pass a value to `done` and result in a failed test